### DISCUSSION TITLE
cast limit & offest to intenger

### DISCUSSION THREAD

**Discussion by jimmywarting** [Automattic/mongoose#12349] [Ideas]
Hi, i was using `URLSearchParams` to fetch data with a `GET` request as a query filter

i didn't want to use json/post cuz i don't save anything and i wanted to take advantage of caching GET requests in browsers too, therefore i used `URLSearchParams` instead

but the thing was that it returns all numbers as a strings instead of numbers so when i did `Collection.find({}).skip(x).limit(y)` then i got a "FailedToParse" MongoDB error. 

the simple solution was to cast them into numbers.
but i was thinking that it would be good if you did use `parseInt` in those two (skip & limit) functions too so that `skip("2")` just works

**Comment by vkarpov15** [comment]
@jimmywarting what version of Mongoose are you using? I checked and we do convert `limit` and `skip` to numbers here: https://github.com/Automattic/mongoose/blob/a792dc3ccfcc73fb8e2227d3ad522b6530c929d0/lib/query.js#L843-L856 and here: https://github.com/Automattic/mongoose/blob/a792dc3ccfcc73fb8e2227d3ad522b6530c929d0/lib/query.js#L877-L890 .

**Reply by jimmywarting** [reply]
^5.12.7

**Reply by vkarpov15** [reply]
@jimmywarting that explains the issue. You need to upgrade to Mongoose 6 for this feature.

**Reply by jimmywarting** [reply]
Gotcha
